### PR TITLE
TE5-T312 Swift Advance Topics

### PR DESCRIPTION
**Pull request by jay-a-shah**

### CODE REVIEW

**Review by purvesh-d — CHANGES_REQUESTED** [review]
- Make single commit
- Code formatting

File: Demo app/Demo app/SecondViewController.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//
+//  Created by Jay Shah on 27/01/22.
+//
+
+import UIKit
+
+protocol TextDelegate {
+    func sendText(text: String)
+}
+class SecondViewController: UIViewController {
```
Comment (purvesh-d) [inline Demo app/Demo app/SecondViewController.swift]: put empty line before starting class

File: Demo app/Demo app/ViewController.swift [hunk 2]
```diff
@@ -8,10 +8,22 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    @IBOutlet weak var textLabel: UILabel!
```
Comment (purvesh-d) [inline Demo app/Demo app/ViewController.swift:12]: Put `//MARK:- Outlets`

File: Demo app/Demo app/ViewController.swift [hunk 3]
```diff
@@ -8,10 +8,22 @@
...
+    @IBOutlet weak var btn1: UIButton!
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
     }
+    @IBAction func goToSecondView(_ sender: UIButton) {
```
Comment (purvesh-d) [inline Demo app/Demo app/ViewController.swift:28]: Create extension for action and put all action func inside it.

File: Swift-Advance-Exercise.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,808 @@
...
+//Create a swift program to demonstrate swift class. Which has two variables inside class body. Access (Set its values and get its value.) these variables by creating one instance of class.
+class One{
+    var var1:Int = 25
+    var var2:String = "Jay"
+}
+var objOne = One()
+print("Both Variables are \(objOne.var1) and \(objOne.var2)")
+
+//Create a swift program to demonstrate usage of swift class for usage of multiple initializers. Create one initializer with two parameters your name and college and another initializer with your name and department. You can be able to access these properties using instance of class.
+class Two{
```
Comment (purvesh-d) [inline Swift-Advance-Exercise.playground/Contents.swift]: `class Two {`

File: Swift-Advance-Exercise.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,808 @@
...
+//Input: [1, 3, 5, 6, 8, 10, 9, 7, 8, 12]
+//Output: even numbers are: [6, 8, 10, 8, 12]
+//                 Odd numbers are: [1, 3, 5, 9, 7]
+struct Arraybiffer {
+    var input = Array<Int>()
+    var evenNumbers = Array<Int>()
+    var oddNumbers = Array<Int>()
+    init(input: [Int]){
+        self.input = input
+        for i in input{
```
Comment (purvesh-d) [inline Swift-Advance-Exercise.playground/Contents.swift]: `for i in input {`

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: Demo app/Demo app/SecondViewController.swift [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+//
+
+import UIKit
+
+protocol TextDelegate {
+    func sendText(text: String)
+}
+
+class SecondViewController: UIViewController {
+//MARK: - Outlet
```
Comment (purvesh-d) [inline Demo app/Demo app/SecondViewController.swift]: Proper indentation

File: Demo app/Demo app/ViewController.swift (hunk 2 above)
Comment (purvesh-d) [reply Demo app/Demo app/ViewController.swift:12]: Proper indentation

File: Swift-Advance.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,522 @@
+import UIKit
+
+
```
Comment (purvesh-d) [inline Swift-Advance.playground/Contents.swift]: Empty extra lines

File: Swift-Advance.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+let fill = Color.blue
+
+
+enum Season {
+    case spring, summer, autumn, winter
+}
+var currentSeason = Season.winter
+print("Current Season:", currentSeason)
+
+enum Laptop{
```
Comment (purvesh-d) [inline Swift-Advance.playground/Contents.swift]: Space requierd

File: Swift-Advance.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+        self.name1 = name1
+        self.cost1 = cost1
+        self.price = price
+        self.totalquantity = cost1*price
+    }
+}
+var objitem = Item1(name1: "Iphone 13", cost1: 102000, price: 2)
+print("\(objitem.totalquantity)")
+
+
```
Comment (purvesh-d) [inline Swift-Advance.playground/Contents.swift]: Extra empty line

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: Swift-Advance-Exercise.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,808 @@
...
+                 let newArray = days.filter{$0.lowercased().contains(index.lowercased())}
+                 return newArray
+             }
+    }
+}
+var p = Arrayofweek()
+print(p["s"])
+
+//14)  Create a swift program to show usage of subscript in string. I pass int argument and it returns the character at given position.
+class Character1 {
```
Comment (purvesh-d) [inline Swift-Advance-Exercise.playground/Contents.swift]: Class name should be proper

File: Swift-Advance-Exercise.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,808 @@
...
+
+// Create a program for shopping cart. If desired quantity for an item is not available, throw exception.
+enum ShoppingCart:Error {
+    case EmptyShoppingCart,Nodesiredquantity
+}
+func ShoppingCartItem(item: Int)throws {
+    if item == 0 {
+        print("No item left")
+        throw ShoppingCart.EmptyShoppingCart
+    }else if item > 10 {
```
Comment (purvesh-d) [inline Swift-Advance-Exercise.playground/Contents.swift:763]: proper spacing

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: Demo app/Demo app/SecondViewController.swift [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+import UIKit
+
+protocol TextDelegate {
+    func sendText(text: String)
+}
+
+class SecondViewController: UIViewController{
+    //MARK: - Outlet
+    @IBOutlet weak var txtfield: UITextField!
+    var delegate : TextDelegate?
```
Comment (Purvesh-Dodiya) [inline Demo app/Demo app/SecondViewController.swift:18]: need mark for variable

File: Demo app/Demo app/SecondViewController.swift [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    
+    override func viewDidLoad() {
```
Comment (Purvesh-Dodiya) [inline Demo app/Demo app/SecondViewController.swift:20]: Need mark

File: Demo app/Demo app/SecondViewController.swift [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+        super.viewDidLoad()
+    }
+}
+extension SecondViewController{
+    //MARK: - Outlet Action
```
Comment (Purvesh-Dodiya) [inline Demo app/Demo app/SecondViewController.swift]: mark should be outside the extension

File: Demo app/Demo app/ViewController.swift [hunk 15]
```diff
@@ -8,10 +8,29 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    //MARK: - Outlet
+    @IBOutlet weak var textLabel: UILabel!
+    @IBOutlet weak var btn1: UIButton!
+    
     override func viewDidLoad() {
```
Comment (Purvesh-Dodiya) [inline Demo app/Demo app/ViewController.swift:15]: same

File: Demo app/Demo app/ViewController.swift [hunk 16]
```diff
@@ -8,10 +8,29 @@
...
         super.viewDidLoad()
         // Do any additional setup after loading the view.
     }
 }
-
+// MARK: - New Part
```
Comment (Purvesh-Dodiya) [inline Demo app/Demo app/ViewController.swift]: give meaning full name to mark

File: Demo app/Demo app/ViewController.swift [hunk 17]
```diff
@@ -8,10 +8,29 @@
...
+    
+    func sendText(text: String) {
+        textLabel.text = text
+    }
+}
+//MARK: - Outlet Action
+extension ViewController {
+  
+    @IBAction func goToSecondView(_ sender: UIButton) {
+      if let controller = storyboard?.instantiateViewController(withIdentifier: "secondViewController") as? SecondViewController {
```
Comment (Purvesh-Dodiya) [inline Demo app/Demo app/ViewController.swift:29]: move identifier to constant file

File: Swift-Advance.playground/Contents.swift (hunk 9 above)
Comment (Purvesh-Dodiya) [reply Swift-Advance.playground/Contents.swift]: pending

**Review by Purvesh-Dodiya — CHANGES_REQUESTED** [review]

File: Demo app/Demo app/SecondViewController.swift (hunk 12 above)
Comment (jay-a-shah) [reply Demo app/Demo app/SecondViewController.swift:18]: Changes Done

**Pull request merged**